### PR TITLE
#152 - new design for dashboard

### LINKED ISSUE #152: New design for dashboard
We should think again about the dashboard. At this moment dashboard for employee with vacation limit looks like this:
image
 
But for users with no vacation limits it looks little worse:

image

But we can improve this view. For example we can make a calendar week view with a switch to month view and calendars should have a button that redirects to create a request page with a filled date. Below it can be the list of vacation requests. On the right side chart with vacation limits may be placed and list on today absences and holidays. And for users with no vacation limit we can hide the chart and show some statistics, for example how many days of absences they have or so.

Example of month view calendar [link](https://tailwindui.com/components/application-ui/data-display/calendars#component-c29139529079ea762f118812bbeaeb9e).

This is only my proposition. I’m open to suggestions.

### PR DESCRIPTION
**Pull request by kamilniemczycki**
This should close #152 

**Changes:**
1. The dashboard has been redesigned
2. Calendar widget added
3. The behaviour of statuses depending on the limit held has been changed.
4. The holiday and remote working widgets are set to on by default

**Presentation:**

https://user-images.githubusercontent.com/1353655/178266644-cb522def-e2b7-4e0f-95c6-f1a064efb904.mp4

### CODE REVIEW

**Review by krzysztofrewak — COMMENTED** [review]
It would be great if this week/month selection would be remembered by the browser.

**Review by Baakoma — DISMISSED** [review]

File: app/Domain/UserVacationStatsRetriever.php [hunk 1]
```diff
@@ -103,6 +103,15 @@ public function getVacationDaysLimit(User $user, YearPeriod $yearPeriod): int
         return $limit ?? 0;
     }
 
+    public function hasVacationDaysLimit(User $user, YearPeriod $yearPeriod): bool
```
Comment (EwelinaLasowy) [inline app/Domain/UserVacationStatsRetriever.php:106]: The method above `getVacationDaysLimit` looks almost exactly the same as this method. The only difference is return type. We shouldn't duplicate code, but think how to use existing code.

**Review by EwelinaLasowy — DISMISSED** [review]

File: app/Domain/UserVacationStatsRetriever.php [hunk 2]
```diff
@@ -103,6 +103,14 @@ public function getVacationDaysLimit(User $user, YearPeriod $yearPeriod): int
...
+    {
+        return $user->vacationLimits()
+            ->whereBelongsTo($yearPeriod)
+            ->first()
+            ?->hasVacation() ?? false;
```
Comment (krzysztofrewak) [inline app/Domain/UserVacationStatsRetriever.php:111]:
```suggestion
            ->first()?->hasVacation() ?? false;
```

This is more readable, I think.

File: resources/js/Shared/Widgets/HomeOfficeList.vue [hunk 3]
```diff
@@ -26,11 +29,28 @@
...
+        :show-description="false"
+      >
+        <template #head>
+          <HomeCityIcon
+            class="flex justify-center"
+            size="48"
+          />
+        </template>
+        <template #title>
+          Nikt nie pracuje z domu
```
Comment (krzysztofrewak) [inline resources/js/Shared/Widgets/HomeOfficeList.vue:43]:
```suggestion
          Nikt nie pracuje zdalnie
```

You don't need to sit at home during remote work. ;)

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]
Maybe we should rename all "home office" occurences to "remote work"?

**Comment by krzysztofrewak** [comment]
`HomeOffice` is still visible in components names.

**Comment by EwelinaLasowy** [comment]
After deploy this command `php artisan toby:rename-home-office-to-remote-work`should be run.

**Review by krzysztofrewak — APPROVED** [review]
Can I deploy it after this PR would be merged? I would not like to foget about running this command.

**Comment by EwelinaLasowy** [comment]
> Can I deploy it after this PR would be merged? I would not like to foget about running this command.

Yes, no problem.

**Review by Baakoma — APPROVED** [review]

**Pull request merged**